buildpack/layertypes/types.go

```diff
@@ -0,0 +1,8 @@
+package layertypes
```

Comment:
I see why this might have been convenient (to avoid import cycles), but I don't love having a separate package just for this type. I wonder if we could make an interface in the buildpack package
```
type Layer interface {
    Metadata() interface{}
    Types() LayerTypes
}

type LayerTypes {
	Build  bool
	Launch bool
	Cache  bool 
}
```
And have the `v0x` packages return something that implements the interface. Then we could have different structs representing the different file schemas, and which would be quite clear. `Encode` and `Decode` could accept a Layer interface type.

Reply:
Unless I miss understood your suggestion, I don't think it's possible.
Tried to make a simple example to show the problem: https://play.golang.org/p/AxPN3bKNx8U
Please let me know if I'm missing something.
In the meantime, I'll leave the code as is.

Reply:
Yeah I screwed up this suggestion a bit. You couldn't have `Layer.Types()` return a `LayerTypes` without defining `LayerTypes` in a third package still. You would need something like:
```
type Layer interface {
    Metadata() interface{}
    IsBuild() bool
    IsLaunch() bool
    IsCache() bool
}
```

Reply:
Alright, after looking at this more closely I realize my suggestion doesn't actually work in this case, it would require some rethinking. I'll resolve this.